### PR TITLE
feat(v3/types): add whole channel category types

### PR DESCRIPTION
**Pull request by Helloyunho**
## About

This PR adds every type in [channel](https://discord.com/developers/docs/resources/channel).

## Status

- [ ] These changes have been tested against Discord API or do not contain API change.
- [ ] This PR includes only documentation changes, no code change.
- [ ] This PR introduces some Breaking changes.

### CODE REVIEW

File: types/src/channels/embed.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+export interface EmbedImagePayload {
+  url?: string;
+  proxy_url?: string;
+  height?: number;
+  width?: number;
+}
+
+export interface EmbedThumbnailPayload extends EmbedImagePayload {}
```
Comment (MierenManz) [inline types/src/channels/embed.ts:26]: wouldn't `export type EmbedThumbnailPayload = EmbedImagePayload` work here?

Comment (Helloyunho) [reply types/src/channels/embed.ts:26]: yeah it would work but i'm not sure whether discord will add some other properties to the thumbnail payload

**Review by MierenManz — APPROVED** [review]
LGTM check the comment. 
It's also for line 33, 47 and 48 of guild.ts

File: types/src/channels/embed.ts (hunk 1 above)
Comment (MierenManz) [reply types/src/channels/embed.ts:26]: We could always change it later. Idk if the linter warns about this. If it does it's prob better to make it a type alias

**Comment by MierenManz** [comment]
Btw would it be possible to have a reference to each type to the docs?
Might be handy for users looking to contribute. Will allow us to update types easierly aswell

File: types/src/channels/embed.ts (hunk 1 above)
Comment (Helloyunho) [reply types/src/channels/embed.ts:26]: hmm yeah linter does warn about it, i'll change those to type alias

**Comment by Helloyunho** [comment]
> Btw would it be possible to have a reference to each type to the docs?
> Might be handy for users looking to contribute. Will allow us to update types easierly aswell

nice

**Comment by MierenManz** [comment]
yunho, is it possible to fix formatting aswell for the check's to be green?

**Review by MierenManz — APPROVED** [review]
LGTM now

File: types/src/channels/base.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// deno-lint-ignore-file camelcase
+
+// https://discord.com/developers/docs/resources/channel#channel-object-channel-types
```
Comment (DjDeveloperr) [inline types/src/channels/base.ts:3]: I think it'd be better to use JSDoc's `@link` attribute for these links, what do you think? Also maybe we could add descriptions for fields based on the API docs?

Comment (Helloyunho) [reply types/src/channels/base.ts:3]: yeah definitely

**Review by DjDeveloperr — APPROVED** [review]
LGTM 🚀

**Pull request merged**